l2geth/core/rollup_fee.go

```diff
@@ -47,34 +54,58 @@ func calculateL1GasLimit(data []byte, overhead uint64) *big.Int {
 	return new(big.Int).SetUint64(gasLimit)
 }
 
-func RoundL1GasPrice(gasPrice uint64) uint64 {
-	if gasPrice == 0 {
-		return gasPrice
-	}
-	if gasPrice == 1 {
-		return hundredMillion
+// ceilModOneHundredMillion rounds the input integer up to the nearest modulus
+// of one hundred million
+func ceilModOneHundredMillion(num *big.Int) *big.Int {
+	if new(big.Int).Mod(num, bigHundredMillion).Cmp(common.Big0) == 0 {
+		return num
 	}
-	if gasPrice%hundredMillion < 2 {
-		gasPrice += hundredMillion - 2
-	} else {
-		gasPrice += hundredMillion
+	sum := new(big.Int).Add(num, bigHundredMillion)
+	mod := new(big.Int).Mod(num, bigHundredMillion)
+	return new(big.Int).Sub(sum, mod)
+}
+
+// VerifyL1GasPrice returns an error if the number is an invalid possible L1 gas
+// price
+func VerifyL1GasPrice(l1GasPrice *big.Int) error {
+	if new(big.Int).Mod(l1GasPrice, bigHundredMillion).Cmp(common.Big0) != 0 {
+		return errInvalidGasPrice
 	}
-	return gasPrice - gasPrice%hundredMillion
+	return nil
 }
 
-func RoundL2GasPrice(gasPrice uint64) uint64 {
-	if gasPrice == 0 {
-		return gasPrice
+// VerifyL2GasPrice returns an error if the number is an invalid possible L2 gas
```

Comment:
`big.Int`s are too verbose but we have to use them because we need the arbitrary precision